File: ios/Plugin/SecureConfigStoragePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SecureConfigStoragePlugin)
public class SecureConfigStoragePlugin: CAPPlugin {
    private let implementation = SecureConfigStorage()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func getValueFromConfig(_ call: CAPPluginCall) {
        let value = call.getString("key") ?? ""
        let result = implementation.getValueFromConfig(value)
        if (result != nil) {
            call.resolve([
                "result": result ?? ""
            ])
        } else {
            call.reject("Something went wrong")
        }
    }
}

File: android/src/main/java/com/paulmojicatech/secureconfigstorage/SecureConfigStoragePlugin.java
package com.paulmojicatech.secureconfigstorage;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SecureConfigStorage")
public class SecureConfigStoragePlugin extends Plugin {

    private SecureConfigStorage implementation = new SecureConfigStorage();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void getValueFromConfig(PluginCall call) {
        String key = call.getString("key");

        // Get key value from configuration
        JSObject ret = new JSObject();
        ret.put("result", implementation.getValueFromConfig("Test Conn String"));
    }
}
